aten/src/ATen/native/cpu/GridSamplerKernel.cpp

```diff
@@ -369,18 +377,31 @@ struct ComputeLocation<scalar_t, GridSamplerPadding::Border, align_corners>
   using ComputeLocationBase<scalar_t, align_corners>::ComputeLocationBase;
 
   inline Vec apply(const Vec &in) const {
-    return clip_coordinates(unnormalize(in));
+    Vec res;
+    if (pixel_coords) {
+      res = in;
+    } else {
+      res = unnormalize(in);
+    }
+    return clip_coordinates(res);
   }
 
   inline std::pair<Vec, Vec> apply_get_grad(const Vec &in) const {
-    Vec res, grad_clip;
-    std::tie(res, grad_clip) = clip_coordinates_get_grad(unnormalize(in));
-    return std::make_pair(res, grad_clip & Vec(scaling_factor));
+    Vec res, grad_clip, grad;
+    if (pixel_coords) {
+      res = in;
+      grad = Vec(1);
```

Comment:
just checking that this is not `Vec(scaling_factor)`

Reply:
The reason that this is `Vec(1)` here is because when the grid is given in pixel coordinates there is no rescaling (the rescaling happens in `unnormalize()`), so the gradient doesn't need to be multiplied by anything.
That's why it gets multiplied by 1 here instead of by `scaling_factor` (which is the scaling factor by which `unnormalize()` rescales).

Same thing with [line 361](https://github.com/pytorch/pytorch/blob/201b0e7ea171d68b0729bcfa5cbdb7e4b6ee4ea2/aten/src/ATen/native/cpu/GridSamplerKernel.cpp#L361) above.